### ISSUE TITLE
Potential issue with support for relative references

### ISSUE THREAD

**Issue by whitlockjc** [swagger-api/swagger-js#517]
The [JSON Pointers (RFC 6901)](https://tools.ietf.org/html/rfc6901) specification says that references that start with a `#` or a `/` are document-local references.  So that means `#/definitions/Pet` and `/definitions/Pet` are both the same reference and reference the same location in the local document.  swagger-js, at least based on its test suite, seems to treat references starting with a `/` as a relative remote reference.  So instead of `/some/path` looking for a node in the local document, swagger-js will attempt to load `/some/path` as an external resource relative to the current root.  This goes against RFC 6901.  I would like to propose that we only support the following approaches to relative references _(All of these examples would resolve a document/path relative to the document loaded)_:
- `./docs/project.json`
- `../docs/project.json`
- `docs/project.json`
- `project.json`

When looking at the Swagger 2.0 Specification, I didn't see this format being used for relative references so I didn't know whether to file this issue there or not.

**Comment by webron** [comment]
Okay, I started writing something long here as this is indeed somewhat baffling. We've used relative refs in a few ways, and your input breaks some :)

Eventually, something didn't add up so I dug a bit further. We're not using JSON Pointers directly but [JSON References](https://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03). If you look at [section-4](https://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03#section-4) which talks about resolution:


>   Resolution of a JSON Reference object SHOULD yield the referenced
   JSON value.  Implementations MAY choose to replace the reference with
   the referenced value.

>   If the URI contained in the JSON Reference value is a relative URI,
   then the base URI resolution MUST be calculated according to
   [RFC3986], section 5.2.  Resolution is performed relative to the
   referring document.

  > If a URI contains a fragment identifier, then the fragment should be
   resolved per the fragment resolution mechansim of the referrant
   document.  If the representation of the referrant document is JSON,
   then the fragment identifier SHOULD be interpreted as a
   [JSON-Pointer].

What I understand from it is that if the URI (full or partial) contains a fragment (`#`), then it would be interpreted as a [JSON Pointer](https://tools.ietf.org/html/rfc6901#section-6). If it's just a relative URI, then the JSON Pointer semantics of [JSON String Representation](https://tools.ietf.org/html/rfc6901#section-5) do not apply and it is read as a relative URI to the document itself.

I could be completely in the wrong direction here, but does this make sense?
